src/eval-frame/iodide-api/file.js

```diff
@@ -0,0 +1,96 @@
+import { getState as reduxStoreGetState } from "../store";
+import { getFiles } from "../../editor/tools/server-tools";
+import sendFileRequestToEditor from "../tools/send-file-request-to-editor";
+import { FETCH_RETURN_TYPES } from "../../editor/state-schemas/state-schema";
+
+const DEFAULT_SAVE_OPTIONS = { overwrite: false };
+
+function isString(argument) {
+  return typeof argument === "string" || argument instanceof String;
+}
+
+function confirmIsString(key, argument) {
+  if (!isString(argument)) {
+    throw new Error(
+      `${key} name must be a string, instead received ${typeof argument}`
+    );
+  }
+}
+
+function confirmIsStringOrUndefined(key, argument) {
+  if (!isString(argument) && argument !== undefined) {
+    throw new Error(`
+${key} name must be a string or undefined, instead received ${typeof argument}`);
+  }
+}
+
+function confirmFetchTypeIsReturnable(fetchType) {
+  if (!FETCH_RETURN_TYPES.includes(fetchType)) {
+    throw new Error(`${fetchType} not a returnable fetch type`);
+  }
+}
+
+export function connectExists(getState) {
+  return function exists(fileName) {
+    confirmIsString("fileName", fileName);
+    const files = getFiles(getState());
+    return files.map(f => f.filename).includes(fileName);
+  };
+}
+
+export function connectList(getState) {
+  return function list() {
+    return getFiles(getState()).map(f => f.filename);
+  };
+}
+
+export function connectLastUpdated(getState) {
+  return function lastUpdated(fileName) {
+    confirmIsString("fileName", fileName);
+    const files = getFiles(getState());
+    const thisFile = files.filter(f => f.filename === fileName);
+    if (!thisFile.length) throw Error(`fileName ${fileName} does not exist`);
+    return new Date(thisFile[0].lastUpdated);
+  };
+}
+
+export function handleResourceLoad(fetchType, variableName) {
+  return file => {
+    if (FETCH_RETURN_TYPES.includes(fetchType) && variableName) {
```

Comment:
`fetchType` has already been validated by your argument validation. this whole function seems like overkill if we use async/await, because then we won't need a terse function to use within a `then`

Reply:
True, I forgot to apply the async / await stuff to `file.js` as you noted, so I'll make sure to remove this

Reply:
though this isn't about validation, but about loading the variable into `window`. all the same this will get refactored